cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -298,7 +298,14 @@ class PyValue {
           value = internal::PyDelta_to_us(dt);
           break;
         case TimeUnit::NANO:
-          value = internal::PyDelta_to_ns(dt);
+          if (internal::IsPandasTimedelta(obj)) {
+            OwnedRef nanos(PyObject_GetAttrString(obj, "nanoseconds"));
+            RETURN_IF_PYERROR();
+            RETURN_NOT_OK(internal::CIntFromPython(nanos.obj(), &value));
+            value += internal::PyDelta_to_ns(dt);
```

Comment:
Should something similar be needed for Timestamp (datetime) ?

Reply:
Yes, just forget to add because bugs were raining.

Reply:
Added support for it.